tests/Unit/Persistence/Json/JsonShoppingListRepositoryTest.php

```diff
@@ -155,4 +155,16 @@ public function testStore(): void
 
         $this->repository->store($list);
     }
+
+    public function testDelete(): void
+    {
+        // Arrange: Given a shopping list
+        // Act: We expect the list to be called on once, using the delete method with the list
+        $this->files
```

Comment:
Required: this test doesn't accurately test the scenario. When a file exists, you're expecting the `exists` method to be called first and return `true`. Then you're expecting `unlink` to be called. 

So the correct code here for the test is:

```php
$this->files
    ->expects($this->once())
    ->method('exists')
    ->with('supplies.json')
    ->willReturn(true);

$this->files
    ->expects($this->once())
    ->method('unlink')
    ->with('supplies.json');

$this->repository->delete('supplies');
```

This is equivalent to:

- Given a shopping list called "supplies" that exists;
- When the list is deleted via the repository;
- Then the file handler unlinks (deletes) the file.

This only tests what happens when deleting a list that exists - we should also test deleting a list that does not exist.

- Given that a shopping list called "supplies" does not exist
- When the list is deleted via the repository
- Then an exception is thrown telling us that the list does not exist.

Have a go at adding that as a new test on your `JsonShoppingListRepositoryTest` class. You'll see that it fails, because in your last commit you deleted code in the JSON repository that was correct.

```php
$this->files
    ->expects($this->once())
    ->method('exists')
    ->with('supplies.json')
    ->willReturn(false);

$this->files
    ->expects($this->never())
    ->method('unlink');

$this->expectException(ShoppingListNotFoundException::class);

$this->repository->delete('supplies');
```

Reply:
I went through this feedback (thank you for this) and it does make sense. However both tests still don't pass. I got confused by what you meant was correct with the code because of the screenshot I sent you last week and I assumed that was right, but I went back in my commits and changed it again. 

My question is since the handler checks if they list exists do I still have to test to see if the list exists in the first test for the JsonShoppingListRepo?

Reply:
Thank you for leaving back this feedback. I looked through it yesterday when it was in the middle of peak hotness and couldn't focus properly. I managed to fix the code so it correlated with the tests and both pass now.